packages/@coorpacademy-components/src/template/skill-detail/all-courses.js

```diff
@@ -59,39 +58,34 @@ const AllCourses = (props, context) => {
       </div>
     ) : null;
 
-  useEffect(() => {
-    setSearchResults(sortBy(course => -getOr(0, ['progress'], course), list));
-  }, [list]);
+  const filteredContent = useMemo(() => {
+    return showCompleted ? list : filter(course => course.progress < 1, list);
+  }, [list, showCompleted]);
+
+  const contentResult = useMemo(() => {
+    return filter(skill => searchValueIncluded(skill.title, searchValue), filteredContent);
+  }, [filteredContent, searchValue]);
 
   const handleSearch = useCallback(
     value => {
       setSearchValue(value);
-      setSearchResults(list.filter(skill => searchValueIncluded(skill.title, value)));
     },
-    [list, setSearchValue, setSearchResults]
+    [setSearchValue]
   );
 
   const handleSearchReset = useCallback(() => {
     setSearchValue('');
-    setSearchResults(list);
-  }, [list, setSearchValue, setSearchResults]);
+  }, [setSearchValue]);
 
   const handleShowCompletedToggle = useCallback(() => {
-    const toggledShowCompleted = !showCompleted;
-    setShowCompleted(toggledShowCompleted);
-    if (toggledShowCompleted) {
-      setSearchResults(list);
-      handleSearchReset();
-    } else {
-      setSearchResults(searchResults.filter(skill => skill.progress < 100));
-    }
-  }, [list, searchResults, showCompleted, setShowCompleted, setSearchResults, handleSearchReset]);
+    setShowCompleted(prevShowCompleted => !prevShowCompleted);
+  }, []);
 
   return (
     <>
       <div className={style.continueLearningWrapper}>
-        <span className={style.continueLearningTitle}>{translate('all_courses')}</span>
-        <span className={style.continueLearningNumber}>{totalCourses}</span>
+        <span className={style.continueLearningTitle}>{translate('all_content')}</span>
+        <span className={style.continueLearningNumber}>{size(contentResult)}</span>
```

Comment:
This will make the total number changes according to the filter and search. Is this what we want?

Reply:
I checked with @wernerdurand, it's the behavior we want 👍